### ISSUE TITLE
InvalidArgumentError

### ISSUE THREAD

**Issue by mageshsridhar** [jianz94/e-lstm-d#1]
I tried to run the code. It is showing this error. Can you please help me with this?

Caused by op 'loss/sequential_3_loss/sub', defined at:
  File "train_model.py", line 37, in <module>
    history = model.train(trainX, trainY)
  File "/Users/mageshsridhar/Downloads/e-lstm-d-master/model.py", line 94, in train
    self.model.compile(optimizer=Adam(lr=FLAGS.learning_rate), loss=self.loss)
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 860, in compile
    sample_weight, mask)
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 459, in weighted
    score_array = fn(y_true, y_pred)
  File "/Users/mageshsridhar/Downloads/e-lstm-d-master/utils.py", line 27, in refined_loss
    return K.mean(K.sum(tf.multiply(weight, K.square(y_true - y_pred)), axis=1), axis=-1)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 865, in binary_op_wrapper
    return func(x, y, name=name)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 2629, in _sub
    result = _op_def_lib.apply_op("Sub", x=x, y=y, name=name)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2630, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1204, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Incompatible shapes: [32,274,274] vs. [32,274,7]
	 [[Node: loss/sequential_3_loss/sub = Sub[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"](_arg_sequential_3_target_0_2, sequential_3/dense_3/Sigmoid)]]

**Comment by jianz94** [comment]
I have test the code again and it runs well. Did you follow the instructions in the README?

**Comment by mageshsridhar** [comment]
Yes. Am I missing any packages or something? I installed all the required dependancies and also in the exact same versions mentioned in the README

**Comment by mageshsridhar** [comment]
Can you please tell me the system configuration and python version? I'm running it in macOS catalina with python 3.6.5

**Comment by jianz94** [comment]
The code is tested in ubuntu 16.04 with python3.6. I don't think the error is caused by these. And I cannot figure out why the last dimension of y_pred is 7.

**Comment by mageshsridhar** [comment]
I tried all possible ways. It's showing the same error everywhere. Can you please help me out?

**Comment by jianz94** [comment]
I can't reproduce the error. Did you set the units of decoder to 274? By the way, you can use self.model.summary() to verify the model's structure.

**Comment by mageshsridhar** [comment]
Yes. I'm running Python3  train_model.py --encoder 128 --lstm 256,256 --decoder 274 --num_epochs 1600 --BETA 10 --learning_rate 0.001

**Comment by jianz94** [comment]
Did you miss the bracket after encoder, lstm and decoder?

**Comment by mageshsridhar** [comment]
Yes. It kept showing error. Now I figured it out. It should be "[128]" and "[256,256]" (with quotes). It kept showing when I executed it without the quotes. Thank you so much
